### ISSUE TITLE
[BUG] no error when component links to itself

### ISSUE THREAD

**Issue by ricochet** [wasmCloud/wadm#505]
When the source and target of a link are the same, the application status should surface an error. This should be considered an invalid link. 

```bash
❯ wash app status test

test@ - Deployed                                                                                            
                                                                                            
  Name                                                             Kind                   Status    
  component1                                                  SpreadScaler   Deployed  
  component1 -(test:foo)-> component1     LinkScaler         Deployed  
  component2                                                 SpreadScaler    Deployed  
  httpserver -(wasi:http)-> component1      LinkScaler         Deployed  
  httpserver                                                     SpreadScaler    Deployed  
  ```

**Comment by brooksmtownsend** [comment]
@ricochet to play devils advocate, is there a legitimate use case for recursion with the same component? I tend to agree that this is probably always invalid (especially if not running with a certain scale, since it would just timeout) but figured before we make it an error it would be good to consider if this is a real use case

**Comment by ricochet** [comment]
I couldn't come up with a reason this would be valid, especially not where the solution had to involve recursion over the same component (vs making another component for fan-out-fan-in even if they have the same logic, both components serve as logically different nodes in the DAG).

If we're looking to replicate linking but distributed, then note that circular linking is disallowed in the component model.

**Comment by brooksmtownsend** [comment]
Yeah, I think that if it's something that's not allowed with just regular composition, then it's not something we want to encourage. Maybe there's a real use case here, but we can address that when it comes up
